File: backend/src/test/java/de/bund/digitalservice/ris/norms/integration/HealthEndpointIntegrationTest.java
package de.bund.digitalservice.ris.norms.integration;

import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.test.web.servlet.MockMvc;

class HealthEndpointIntegrationTest extends BaseIntegrationTest {

  @Test
  void actuatorHealthShouldBeReachable(@Autowired MockMvc mvc) throws Exception {
    mvc.perform(get("/actuator/health")).andExpect(status().isOk());
  }
}

File: time-machine/src/integrationTest/kotlin/de/bund/digitalservice/template/HealthEndpointIntegrationTest.kt
package de.bund.digitalservice.template

import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.boot.test.context.SpringBootTest.WebEnvironment
import org.springframework.test.web.reactive.server.WebTestClient

@SpringBootTest(webEnvironment = WebEnvironment.RANDOM_PORT)
class HealthEndpointIntegrationTest(
    @Autowired val webTestClient: WebTestClient,
) {
    @Test
    fun `should expose health endpoint`() {
        webTestClient.get().uri("/actuator/health").exchange().expectStatus().isOk()
    }
}
